marlowe-test/stress-testing.md

```diff
@@ -0,0 +1,111 @@
+                           Wait specified seconds after transaction
+                           confirmation. No waiting occurs if a non-positive
+                           number of seconds is specified. The specified wait
+                           period is randomly increased up to a factor of two.
+                           Increasing this value will increase the probability
+                           that the submitting node has seen the transactions
+                           that Marlowe Runtime has seen. (default: 3)
+  --retry-seconds INTEGER  Wait specified seconds after after a failed
+                           transaction before trying again. No retries occur if
+                           a non-positive number of seconds is specified.
+                           (default: 10)
+  --retry-limit INTEGER    Maximum number of attempts for trying a failed
+                           transaction again. Each subsequent retry waits twice
+                           as long as the previous retry. No retries occur if a
+                           non-positive number of retries is specified.
+                           (default: 5)
+  NATURAL                  The number of contracts to run sequentially for each
+                           party.
+  ADDRESS=KEYFILE          The addresses of the parties and the files with their
+                           signing keys.
+```
+Setting `--build-seconds` and `--confirm-seconds` to zero creates the maximum rate of requests to `marlowe-tx`. Increasing the number of addresses running contract in parallel proportionally increases the number of requests on the backend.
+
+We have observed that `marlowe-scaling` achieves greater than 99.99% success when running 20 contracts (addresses) in parallel against a single Marlowe Runtime backend, if the default settings for delays and retries are used. Note that the current protocol parameters for public Cardano networks make possible running approximately 10 Marlowe transactions per block (i.e., approximately 30 per minute), so attempting to run even 20 Marlowe contracts in parallel as quickly as possible exceeds the Plutus budget for a Cardano block.
+
+
+## Summary Recommendations
+
+Empirical observations indicate that a single Marlowe Runtime backend scales to handle the following well:
+- Wait 6 seconds after the previous confirmation of a transaction before building another Marlowe transaction that is based on the addresses used in the previous transaction. (There is no need to wait after the transactions at irrelevant addresses.)
+- Wait 10 seconds before retrying a failed Marlowe transactions.
+- Retry up to 5 times, doubling the delay from each subsequent retry.
+- Limit (throttle) the demand for building and executing Marlowe transactions on Marlowe Runtime to no more than 20 simultaneous request.
+
+Also note that request to Marlowe Runtime are stateless, so that any request can be routed to any instance of Marlowe Runtime.
```

Comment:
Suggest 'requests to Marlowe Runtime are ...'

Reply:
Fixed in 908a7e461d7bc492d89665f3ab46ef743ea02c76